diff --git a/dist/chunk-ABQEMJ44.cjs b/dist/chunk-ABQEMJ44.cjs
index ea30c56523fd507559d41a2329e6f1027faaecc6..1b1e8130e2fcd699c8f586e3c66a50df126a4220 100644
--- a/dist/chunk-ABQEMJ44.cjs
+++ b/dist/chunk-ABQEMJ44.cjs
@@ -36,7 +36,7 @@ var TypeScriptEslintComplexityAdapter = class {
     const ast = _typescriptestree.parse.call(void 0, sourceText, {
       loc: true,
       range: true,
-      jsx: true
+      jsx: false
     });
     const results = [];
     this.walkProgram(ast, filePath, results);
diff --git a/dist/chunk-RBS3BIJZ.js b/dist/chunk-RBS3BIJZ.js
index d10b12534449fd6f2f895204f42723072efe79e4..0627b47d2d78e04f6aed4abee2779e09c25921b5 100644
--- a/dist/chunk-RBS3BIJZ.js
+++ b/dist/chunk-RBS3BIJZ.js
@@ -36,7 +36,7 @@ var TypeScriptEslintComplexityAdapter = class {
     const ast = parse(sourceText, {
       loc: true,
       range: true,
-      jsx: true
+      jsx: false
     });
     const results = [];
     this.walkProgram(ast, filePath, results);
